packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/PropertyProvider.cs

```diff
@@ -59,6 +68,54 @@ public PropertyProvider(InputModelProperty inputProperty)
             ExplicitInterface = explicitInterface;
             XmlDocs = GetXmlDocs();
             WireInfo = wireInfo;
+
+            InitializeParameter(Name, description ?? FormattableStringHelpers.Empty, Type, ParameterValidationType.None);
+        }
+
+        [MemberNotNull(nameof(_parameter))]
+        [MemberNotNull(nameof(_inputParameter))]
+        private void InitializeParameter(string propertyName, FormattableString description, CSharpType propertyType, ParameterValidationType validation)
+        {
+            var parameterName = propertyName.FirstCharToLowerCase();
```

Comment:
I see we use ToVariableName in many other places. Can you explain why we use this new method here?

Reply:
well those methods should come with a pair - ToLowerCase vs ToUpperCase, and ToVariableName vs ???
We do not yet have that pair for `ToVariableName` yet, because we are constructing the property names by `FirstCharToUpperCase`.

Reply:
well I think it should `ToCleanName`.
When we are building the property names, we should call `ToCleanName`.
And when we are building the parameter names, we should call `ToVariableName`.
Either of them should be `FirstCharToLowerCase` or ToUpperCase. For instance, the raw property name could have hyphen or  underscore in them, and we need to clean it up.